### PR TITLE
Migrate Consent Portal to use direct Google Auth + RFC 8693 Token Exchange

### PR DESCRIPTION
**Pull request by ginaxu1**
## Summary
Migrate the Consent Portal SPA authentication away from the legacy OIDC redirect flow (which failed with `FET-1002` in newer ThunderID versions) to direct Google login combined with token exchange, aligning it with the pattern successfully used by the Passport application

## Type of Change
- [x] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)

### 1. Consent Portal Auth Migration
- Auth context & Google sign-in integration: Leveraged `@react-oauth/google` for local Google token retrieval and created `AuthContext.tsx` to handle direct OIDC token exchange (`urn:ietf:params:oauth:grant-type:token-exchange`) with ThunderID using `client_id=CONSENT_PORTAL_APP`.
- Cleaned legacy imports: Entirely purged `react-oidc-context` and `oidc-client-ts` dependencies. Migrated `UserHeader.tsx`, `ErrorPage.tsx`, `StatusInfoPage.tsx`, `SuccessPage.tsx`, and `UnauthorizedPage.tsx` to consume the custom `AuthContext`.
- Config & Formatting: Exposed `googleClientId` and `thunderIdTokenUrl` on the runtime configuration, updated `config.example.js`, and ran `prettier` code formatting.

## Testing

- [x] I have tested this change locally
- [ ] I have added unit tests for new functionality
- [x] I have tested edge cases
- [x] All existing tests pass

## Checklist

- [x] My code follows the project's style guidelines
- [x] I have performed a self-review of my code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have checked that there are no merge conflicts

## Additional Notes

ThunderID's configuration in the target environment has been updated to explicitly accept the OIDC token exchange grant type `urn:ietf:params:oauth:grant-type:token-exchange` for the consent portal client.

### CODE REVIEW

File: portals/consent-portal/package.json [hunk 1]
```diff
@@ -10,12 +10,11 @@
     "preview": "vite preview"
   },
   "dependencies": {
+    "@react-oauth/google": "^0.13.5",
```
Comment (mushrafmim) [inline portals/consent-portal/package.json:13]: We can't use a vendor specific Auth for the consent portal, this is a generic product and no country is gonna use Google Auth as their digital ID.

We should still stick to `react-oidc-context` while adding the Federated Google Auth to the Farajaland.

**Review by mushrafmim — CHANGES_REQUESTED** [review]

File: portals/consent-portal/package.json (hunk 1 above)
Comment (ginaxu1) [reply portals/consent-portal/package.json:13]: Fair point @mushrafmim , closing this PR then since `main` already uses `react-oidc-context`, no changes necessary in this repo.

For Farajaland, the federated Google login is already handled by ThunderID (it's configured as an OIDC Identity Provider in `02-custom-resources.yaml` and wired into the default authentication flow. So when the consent portal redirects to ThunderID via standard OIDC, ThunderID handles the Google sign-in on its end. The portal stays completely vendor-agnostic

PR for `opendif-farajaland` https://github.com/OpenNDX/openndx-farajaland/pull/44/changes
